docs/source/kinds_of_types.rst

```diff
@@ -721,18 +721,31 @@ For more details, see :ref:`type-variable-value-restriction`.
 Generators
 **********
 
-A basic generator that only yields values can be annotated as having a return
+A basic generator that only yields values can be succinctly annotated as having a return
 type of either :py:class:`Iterator[YieldType] <typing.Iterator>` or :py:class:`Iterable[YieldType] <typing.Iterable>`. For example:
 
 .. code-block:: python
 
    def squares(n: int) -> Iterator[int]:
        for i in range(n):
            yield i * i
+           
+In general, :py:class:`Iterator[YieldType] <typing.Iterator>` should be preferred over
+:py:class:`Iterable[YieldType] <typing.Iterable>` for the return type of a generator function,
+as it is more precise regarding the type of the returned object. All iterator objects have a
+:py:meth:`~iterator.__next__` method, so using ``Iterator`` as the return type alerts mypy to
+the fact that :py:func:`next` can be called on the object returned from the generator function.
+Using ``Iterable`` runs the risk of mypy raising false-positive errors if a user attempts to call
+`next()` on the returned object.
+
+However, in some cases, it may be considered an implementation detail that ``next()`` can be
+called on the object returned from the generator function. In these cases, it may be preferable
+to use ``Iterable`` instead of ``Iterator``.
 
 If you want your generator to accept values via the :py:meth:`~generator.send` method or return
-a value, you should use the
-:py:class:`Generator[YieldType, SendType, ReturnType] <typing.Generator>` generic type instead. For example:
+a value, meanwhile, you should use the
```

Comment:
```suggestion
a value, you should use the
```

Reply:
Sorry to be a pain, but I actually quite liked the word "meanwhile" here! I think it creates a better flow between the paragraphs, more clearly indicating that *this* situation is in contrast to the *previous* situation discussed in the previous paragraph.

I'd be open to other words such as "however" 🙂

Reply:
Okay, makes sense, I put in "however"!
(to explain my original nit here: I didn't originally read that "meanwhile" as referring to a previous paragraph, and without that context "meanwhile" connotes a kind of temporariness or incompleteness)

Reply:
I think the reason I went for "meanwhile" over "however" was that I'd used "however" just two sentences prior, fwiw 🙂 (I'm happy with "however", so don't want to belabor the point!!)